### PR TITLE
feat(acm)!: remove direct kubectl commands

### PR DESCRIPTION
**Pull request by foosinn**
kubectl-wrapper currently breaks if one has to access the api using a proxy (IAP).

This replaces the wrapper call for all annotations with the `kubernetes_annotations` resource. The `kubernetes` provider is already in use for the secret.

In addition it allows the user to disable the automated restart. Sadly it's currently not possible to work around this AFAIK

### CODE REVIEW

**Comment by foosinn** [comment]
fixed the docs

**Comment by foosinn** [comment]
@apeabody I've got some feedback on the kubernetes restart handler:

https://github.com/hashicorp/terraform-provider-kubernetes/issues/2304#issuecomment-1748971887

Would you prefer to leave it as is, or should i update the PR for the recommended solution? I'm not really sure of the use of `time_static` would have any implications to your release cycle

**Comment by apeabody** [comment]
> @apeabody I've got some feedback on the kubernetes restart handler:
> 
> [hashicorp/terraform-provider-kubernetes#2304 (comment)](https://github.com/hashicorp/terraform-provider-kubernetes/issues/2304#issuecomment-1748971887)
> 
> Would you prefer to leave it as is, or should i update the PR for the recommended solution? I'm not really sure of the use of `time_static` would have any implications to your release cycle

Hi @foosinn - This will be going into major version (https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1746), so I would recommend including everything in this PR so any upgrade migration required is only one-time.  It appears we already include the `time` provider as a [requirement](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/acm/versions.tf#L54).  Any required upgrade documentation can be added as `docs/upgrading_to_v29.0.md`

**Comment by foosinn** [comment]
i'll update tomorrow to the recommended fix.

Can you somehow share the log output of `Build 5d254629-88a8-47ca-bc26-da74d399ea9f failed`? Seems like it's not public

**Comment by foosinn** [comment]
Switched to the suggested solution and also rebased latest master. Let me know if you need anything else.

Did also a quick test with our fork, seems to work fine :)

**Comment by apeabody** [comment]
From the linter

```
Checking submodule's files generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/acm/README.md /tmp/tmp.vcv4AgP5L4/workspace/modules/acm/README.md
98d97
< | restart\_gatekeeper\_controller\_manager | Restart the gatekeeper controller manager after setting up workload id (needs to be done manually if a proxy to gke api is required) | `bool` | `true` | no |
Error: submodule's files generation has not been run, please run the
'make build' command and commit changes
/usr/local/bin/task_helper_functions.sh: line 30: DELETE_AT_EXIT: readonly variable
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=autogen' '--exclude=*.tfvars' '--exclude=*metadata.yaml' /workspace/modules/acm/README.md /tmp/tmp.ONYkjFiaEt/generate_docs/workspace/modules/acm/README.md
98d97
< | restart\_gatekeeper\_controller\_manager | Restart the gatekeeper controller manager after setting up workload id (needs to be done manually if a proxy to gke api is required) | `bool` | `true` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
```

**Comment by foosinn** [comment]
And again fixed the docs

File: modules/acm/creds.tf [hunk 1]
```diff
@@ -39,63 +39,65 @@ resource "time_sleep" "wait_acm" {
...
-  source  = "terraform-google-modules/gcloud/google//modules/kubectl-wrapper"
-  version = "~> 3.1"
-
-  count            = var.enable_policy_controller && var.create_metrics_gcp_sa ? 1 : 0
-  skip_download    = true
-  cluster_name     = var.cluster_name
-  cluster_location = var.location
-  project_id       = var.project_id
+resource "time_static" "restarted_at" {}
+resource "kubernetes_annotations" "annotate-sa-gatekeeper-system-restart" {
```
Comment (apeabody) [inline modules/acm/creds.tf:89]: Looking at the Int test, this probably needs to `depends_on = [kubernetes_annotations.annotate-sa-gatekeeper-system]`.

Comment (foosinn) [reply modules/acm/creds.tf:89]: fixed

**Comment by foosinn** [comment]
Can you share the error that happened?

**Review by apeabody — APPROVED** [review]
Thanks for the contribution @foosinn 

Marking as a breaking change given the config "churn" window.

**Comment by foosinn** [comment]
thanks!

**Pull request merged**